### PR TITLE
Update auto-install.sh to support Debian 12

### LINKED ISSUE #343: [feature] Support Debian 12
**Is your feature request related to a problem? Please describe.**
Right now docker OpenWISP only supports Debian releases that are either EOL or LTS. Ideally it should support Debian 12 as there is not technical reason not to.

**Describe the solution you'd like**
A minor change to the autoinstall script to support Debian 12. I was able to get it to work. However, because Debian and other distros don't allow managing the system with pip it will be necessary to remove the pip upgrade. I don't really see the need for a pip upgrade anyway.

### PR DESCRIPTION
**Pull request by Darin755**
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #343

## Description of Changes

I removed the upgrade to pip and I added 12 to the supported versions

### CODE REVIEW

File: deploy/auto-install.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ error_msg_with_continue() {
 apt_dependenices_setup() {
 	start_step "Setting up dependencies..."
 	apt --yes install python3 python3-pip git python3-dev gawk libffi-dev libssl-dev gcc make &>>$LOG_FILE
-	pip3 install --upgrade pip &>>$LOG_FILE
+	#pip3 install --upgrade pip &>>$LOG_FILE
```
Comment (nemesifier) [inline deploy/auto-install.sh:58]: Why this change?

Comment (Darin755) [reply deploy/auto-install.sh:58]: APT now manages the python environment not pip

Comment (nemesifier) [reply deploy/auto-install.sh:58]: Let's just remove the line.
```suggestion
```

**Review by nemesifier — CHANGES_REQUESTED** [review]

**Review by nemesifier — APPROVED** [review]
